hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieAppendHandle.java

```diff
@@ -129,6 +129,9 @@ public class HoodieAppendHandle<T, I, K, O> extends HoodieWriteHandle<T, I, K, O
   private boolean useWriterSchema = false;
 
   private Properties recordProperties = new Properties();
+  // Block Sequence number will be used to detect duplicate log blocks(by log reader) added due to spark task retries.
+  // It should always start with 0 for a given file slice. for roll overs and delete blocks, we increment by 1.
+  private int blockSequenceNumber = 0;
```

Comment:
Will this not get reinitialized when spark task is retried ? class is reinstantiated i think

Reply:
Ignore this , did not see attemptNo logic .